sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,228 @@
+#title          :discretization.py
+#description    :Implementation of a few discretization algorithms
+#author         :Henry Lin <hlin117@gmail.com>
+#==============================================================================
+
+from __future__ import division
+from math import log
+import numpy as np
+from scipy.stats import entropy
+from sklearn.base import BaseEstimator
+
+class MDLP(BaseEstimator):
+    """Implements the MDLP discretization criterion from Usama Fayyad's
+    paper "Multi-Interval Discretization of Continuous-Valued Attributes
+    for Classification Learning"
+    """
+
+    def __init__(self, **params):
```

Comment:
Interesting that you've interpreted the contributors' guide this way... All our estimators have an explicit list of parameters, not `**` magic. With an explicit list of parameters, the implementation of `get_params` and `set_params` in `BaseEstimator` will suffice.